### PR TITLE
Overriding the default confirm behaviour - Make the example work

### PR DESCRIPTION
**Pull request by Neophen**
Add a working example of overriding the `data-confirm` behaviour

### CODE REVIEW

File: lib/phoenix_component.ex [hunk 1]
```diff
@@ -2869,7 +2887,7 @@ defmodule Phoenix.Component do
       "the optional override for the accept attribute. Defaults to :accept specified by allow_upload"
   )
 
-  attr.(:rest, :global, include: ~w(webkitdirectory required disabled capture form))
+  attr.(:rest, :global, include: ~w(webkitdirectory required disabled))
```
Comment (chrismccord) [inline lib/phoenix_component.ex:2890]:
```suggestion
  attr.(:rest, :global, include: ~w(webkitdirectory required disabled capture form))
```

**Comment by chrismccord** [comment]
Which part about the existing docs didn't work? Also please keep type="button". Thanks!

**Comment by Neophen** [comment]
Yeah not sure how that changed will fix


Here's the more in depth [article](https://dev.to/neophen/beter-data-confirm-modals-in-phoenix-liveview-5al5), but overall the `window.confirm/prompt/alert` are blocking browser behaviour so it feels like you can `event.preventDefault()` after a modal, but in reality it only works with the `window.confirm/alert/prompt as they have browser implementation to stop executing.

hence the re-trigger behaviour required.

**Comment by Neophen** [comment]
@chrismccord not sure if need to ping at all, the ci seems to fail for no reason.

File: lib/phoenix_component.ex [hunk 2]
```diff
@@ -2572,16 +2572,35 @@ defmodule Phoenix.Component do
...
     // Prevent default implementation
     e.stopPropagation();
     // Introduce alternative implementation
     var message = e.target.getAttribute("data-confirm");
     if(!message){ return true; }
+
+    // Confirm is resolved execute the click event
+    if (e.target?.hasAttribute(RESOLVED_ATTRIBUTE)) {
+        e.target.removeAttribute(RESOLVED_ATTRIBUTE);
+        return true;
```
Comment (SteffenDE) [inline lib/phoenix_component.ex:2587]:
```suggestion
      e.target.removeAttribute(RESOLVED_ATTRIBUTE);
      return;
```

File: lib/phoenix_component.ex [hunk 3]
```diff
@@ -2572,16 +2572,35 @@ defmodule Phoenix.Component do
...
     vex.dialog.confirm({
       message: message,
       callback: function (value) {
-        if (value == false) { e.preventDefault(); }
+        if (value == true) {
+            // Customer confirmed, re-trigger the click event.
+            e.target?.click();
+        } else {
+            // Customer canceled
+            e.target?.removeAttribute(RESOLVED_ATTRIBUTE);
```
Comment (SteffenDE) [inline lib/phoenix_component.ex:2602]:
```suggestion
          // Customer canceled
          e.target?.removeAttribute(RESOLVED_ATTRIBUTE);
```

File: lib/phoenix_component.ex [hunk 4]
```diff
@@ -2572,16 +2572,35 @@ defmodule Phoenix.Component do
...
+    e.preventDefault();
+    e.target?.setAttribute(RESOLVED_ATTRIBUTE, "");
+
     vex.dialog.confirm({
       message: message,
       callback: function (value) {
-        if (value == false) { e.preventDefault(); }
+        if (value == true) {
+            // Customer confirmed, re-trigger the click event.
+            e.target?.click();
```
Comment (SteffenDE) [inline lib/phoenix_component.ex:2599]:
```suggestion
          // Customer confirmed, re-trigger the click event.
          e.target?.click();
```

File: lib/phoenix_component.ex [hunk 5]
```diff
@@ -2572,16 +2572,35 @@ defmodule Phoenix.Component do
...
   // listen on document.body, so it's executed before the default of
   // phoenix_html, which is listening on the window object
+  const RESOLVED_ATTRIBUTE = "data-confirm-resolved";
+
   document.body.addEventListener('phoenix.link.click', function (e) {
     // Prevent default implementation
     e.stopPropagation();
     // Introduce alternative implementation
     var message = e.target.getAttribute("data-confirm");
     if(!message){ return true; }
```
Comment (SteffenDE) [inline lib/phoenix_component.ex:2582]:
```suggestion
    if(!message){ return; }
```

File: lib/phoenix_component.ex [hunk 6]
```diff
@@ -2572,16 +2572,35 @@ defmodule Phoenix.Component do
   ```javascript
   // listen on document.body, so it's executed before the default of
   // phoenix_html, which is listening on the window object
+  const RESOLVED_ATTRIBUTE = "data-confirm-resolved";
+
   document.body.addEventListener('phoenix.link.click', function (e) {
```
Comment (SteffenDE) [inline lib/phoenix_component.ex:2577]:
```suggestion
  // Compared to a javascript window.confirm, the custom dialog does not block
  // javascript execution. Therefore to make this work as expected we store
  // the successful confirmation as an attribute and re-trigger the click event.
  // On the second click, the `data-confirm-resolved` attribute is set and we proceed.
  const RESOLVED_ATTRIBUTE = "data-confirm-resolved";
  // listen on document.body, so it's executed before the default of
  // phoenix_html, which is listening on the window object
  document.body.addEventListener('phoenix.link.click', function (e) {
```

**Comment by SteffenDE** [comment]
Thank you! 🙌🏻

I'll try to add an e2e test for this separately.

**Pull request merged**